src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -525,15 +525,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
 
   <Target Name="RenamePublishedHost"
+          DependsOnTargets="_ComputeNETCoreBuildOutputFiles"
           AfterTargets="ComputeFilesToPublish"
           BeforeTargets="CopyFilesToPublishDirectory"
           Condition="'$(RenamePublishedHost)' == 'true'">
 
     <ItemGroup>
 
-      <ResolvedFileToPublish Condition="'%(ResolvedFileToPublish.RelativePath)' == '$(_DotNetHostExecutableName)'">
-        <RelativePath>$(AssemblyName)%(ResolvedFileToPublish.Extension)</RelativePath>
-      </ResolvedFileToPublish>
+      <ResolvedFileToRemove  Include ="%(ResolvedFileToPublish.Identity)" Condition="'%(ResolvedFileToPublish.RelativePath)' == '$(_DotNetHostExecutableName)'"/>
+      <ResolvedFileToPublish Remove ="%(ResolvedFileToRemove.Identity)"/>
+
+      <ResolvedFileToAdd Include="%(NativeAppHostNETCore.Identity)">
```

Comment:
Why use 2 items here?  Why not just Include this into `<ResolvedFileToPublish>`?